collection/service.go

```
@@ -147,24 +149,40 @@ func addCollectionItemQuery(joinedLabels string, relation string, contentCollect
 
 //Delete - Deletes a content collection
 func (pcd service) Delete(uuid string) (bool, error) {
+	var queries []*neoism.CypherQuery
+
 	removeRelationships := &neoism.CypherQuery{
-		Statement: fmt.Sprintf(`MATCH (n:Thing {uuid: {uuid}})
-			OPTIONAL MATCH (item:Thing)<-[rel:%s]-(n)
+		Statement: fmt.Sprintf(`MATCH (cc:Thing {uuid: {uuid}})
+			OPTIONAL MATCH (item:Thing)<-[rel:%s]-(cc)
 			DELETE rel`, pcd.relation),
 		Parameters: map[string]interface{}{
 			"uuid": uuid,
 		},
 	}
+	queries = append(queries, removeRelationships)
+
+	if pcd.extraRelForDelete != "" {
+		removeExtraRelationships := &neoism.CypherQuery{
+			Statement: fmt.Sprintf(`MATCH (cc:Thing {uuid: {uuid}})
+				OPTIONAL MATCH (t:Thing)<-[rel:%s]-(cc)
+				DELETE rel`, pcd.extraRelForDelete),
+			Parameters: map[string]interface{}{
+				"uuid": uuid,
+			},
+		}
+		queries = append(queries, removeExtraRelationships)
+	}
 
 	removeNode := &neoism.CypherQuery{
-		Statement: `MATCH (n:Thing {uuid: {uuid}}) DELETE n`,
+		Statement: `MATCH (cc:Thing {uuid: {uuid}}) DELETE cc`,
 		Parameters: map[string]interface{}{
 			"uuid": uuid,
 		},
 		IncludeStats: true,
 	}
+	queries = append(queries, removeNode)
 
-	err := pcd.conn.CypherBatch([]*neoism.CypherQuery{removeRelationships, removeNode})
+	_ = pcd.conn.CypherBatch(queries)
```

I think we should handle errors for the whole batch execution (I see the error was ignored before as well). Does anyone know if there was any particular reason why we ignored this?